aws/resource_aws_media_convert_queue.go

```diff
@@ -0,0 +1,264 @@
+		Read:   resourceAwsMediaConvertQueueRead,
+		Update: resourceAwsMediaConvertQueueUpdate,
+		Delete: resourceAwsMediaConvertQueueDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"pricing_plan": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  mediaconvert.PricingPlanOnDemand,
+				ValidateFunc: validation.StringInSlice([]string{
+					mediaconvert.PricingPlanOnDemand,
+					mediaconvert.PricingPlanReserved,
+				}, false),
+			},
+			"reservation_plan_settings": {
+				Type:     schema.TypeList,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"commitment": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								mediaconvert.CommitmentOneYear,
+							}, false),
+						},
+						"renewal_type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								mediaconvert.RenewalTypeAutoRenew,
+								mediaconvert.RenewalTypeExpire,
+							}, false),
+						},
+						"reserved_slots": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  mediaconvert.QueueStatusActive,
+				ValidateFunc: validation.StringInSlice([]string{
+					mediaconvert.QueueStatusActive,
+					mediaconvert.QueueStatusPaused,
+				}, false),
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsMediaConvertQueueCreate(d *schema.ResourceData, meta interface{}) error {
+	originalConn := meta.(*AWSClient).mediaconvertconn
+
+	endpointURL, err := getAwsMediaConvertEndpoint(originalConn)
+	if err != nil {
+		return fmt.Errorf("Error getting Media Convert Endpoint: %s", err)
+	}
+
+	sess, err := session.NewSession(&originalConn.Config)
+	if err != nil {
+		return fmt.Errorf("Error creating AWS session: %s", err)
+	}
+	conn := mediaconvert.New(sess.Copy(&aws.Config{Endpoint: aws.String(endpointURL)}))
+
+	createOpts := &mediaconvert.CreateQueueInput{
+		Name:        aws.String(d.Get("name").(string)),
+		Status:      aws.String(d.Get("status").(string)),
+		PricingPlan: aws.String(d.Get("pricing_plan").(string)),
+		Tags:        tagsFromMapGeneric(d.Get("tags").(map[string]interface{})),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		createOpts.Description = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("reservation_plan_settings"); ok {
+		reservationPlanSettings := v.([]interface{})[0].(map[string]interface{})
```

Comment:
To prevent potential panics, we should perform length and nil checks, e.g.

```go
if v, ok := d.Get("reservation_plan_settings").([]interface{}); ok && len(v) > 0 && v[0] != nil {
```

Or perform that logic in `expandMediaConvertReservationPlanSettings()` and accept the `[]interface{}` as the argument instead. 👍